### PR TITLE
feature: add SONATA support

### PR DESCRIPTION
**Pull request by matz-e**
Introduce a new virtual parent class for `MVD3File` and the new `SonataFile`, providing a (currently) needed set of attributes to extract from both file types.

Notable changes:
- `MVD::SonataFile` has been added
- `MVD::open` is the new entry point, will be able to open MVD3 and SONATA files. Rudimentary functionality to test that the files can be opened.
- deprecate older entry points, which did not test if files could be opened

Dependent projects will have to be adjusted slightly to take advantage of the new abilities (`MVD3File` should be replaced with `shared_ptr<MVD::File>`, and `MVD::open` used for file opening).

Still missing:
- [x] improve unit test coverage
- [ ] ensure all needed attributes are covered (add `me_type` and SONATA equivalent)
- [x] update Spack dependencies for CI

Depends on BlueBrain/libsonata#39

### CODE REVIEW

File: include/mvd/mvd_generic.hpp [hunk 1]
```diff
@@ -42,6 +44,42 @@ inline MVDFile* open_mvd(const std::string & filename) {
...
+/// Opens a file with the associated class. A basic check for file type is
+/// performed, but no guarantees about file correctness are made.
+///
+/// \param filename the path of the file to open
+/// \return a shared pointer to a mvd::File object
+///
+inline std::shared_ptr<File> open(const std::string& filename) {
+    const std::string mvd3 = ".mvd3";
+
+    std::cout << filename << "\n";
```
Comment (tristan0x) [inline include/mvd/mvd_generic.hpp]: I am not sure this `std::cout << ...` is wanted.

Comment (matz-e) [reply include/mvd/mvd_generic.hpp]: Ooops… yes

File: include/mvd/mvd_base.hpp [hunk 2]
```diff
@@ -60,11 +61,39 @@ class MVDFile {
 };
 
 
+///
+/// \brief Higher level interface for functional access
+///
+class File : public MVDFile {
+public:
+    inline File() {}
```
Comment (tristan0x) [inline include/mvd/mvd_base.hpp]:
```suggestion
    inline File() = default;
```

File: include/mvd/mvd_base.hpp [hunk 3]
```diff
@@ -60,11 +61,39 @@ class MVDFile {
...
+    inline virtual ~File() {}
```
Comment (tristan0x) [inline include/mvd/mvd_base.hpp]:
```suggestion
    inline virtual ~File() = default;
```

File: include/mvd/sonata.hpp [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+    /// throw MVDException, or HighFive::Exception in case of error
+    ///
+    SonataFile(const std::string& filename, const std::string& population = "default");
+
+
+    ///
+    /// \brief getNbNeuron
+    /// \return total number of neurons contained in the receipe
+    ///
+    size_t getNbNeuron() const override { return size_; }
```
Comment (tristan0x) [inline include/mvd/sonata.hpp]:
```suggestion
    size_t getNbNeuron() const noexcept override { return size_; }
```

**Comment by matz-e** [comment]
CI build is fixed. One last review?

**Pull request merged**